cmd/cli/pipelines_init.go

```diff
@@ -0,0 +1,335 @@
+)
+
+//go:embed pipeline.tmpl
+var pipelineCfgTmpl string
+
+var funcMap = template.FuncMap{
+	"formatParameterValueTable":      formatParameterValueTable,
+	"formatParameterValueYAML":       formatParameterValueYAML,
+	"formatParameterDescriptionYAML": formatParameterDescriptionYAML,
+	"formatParameterRequired":        formatParameterRequired,
+}
+
+func formatParameterRequired(param config.Parameter) string {
+	for _, v := range param.Validations {
+		if v.Type() == config.ValidationTypeRequired {
+			return "Required"
+		}
+	}
+
+	return "Optional"
+}
+
+// formatParameterValue formats the value of a configuration parameter.
+func formatParameterValueTable(value string) string {
+	switch {
+	case value == "":
+		return `<Chip label="null" />`
+	case strings.Contains(value, "\n"):
+		// specifically used in the javascript processor
+		return fmt.Sprintf("\n```js\n%s\n```\n", value)
+	default:
+		return fmt.Sprintf("`%s`", value)
+	}
+}
+
+func formatParameterDescriptionYAML(description string) string {
+	const (
+		indentLen  = 10
+		prefix     = "# "
+		lineLen    = 80
+		tmpNewLine = "〠"
+	)
+
+	// remove markdown new lines
+	description = strings.ReplaceAll(description, "\n\n", tmpNewLine)
+	description = strings.ReplaceAll(description, "\n", " ")
+	description = strings.ReplaceAll(description, tmpNewLine, "\n")
+
+	formattedDescription := formatMultiline(description, strings.Repeat(" ", indentLen)+prefix, lineLen)
+	// remove first indent and last new line
+	formattedDescription = formattedDescription[indentLen : len(formattedDescription)-1]
+	return formattedDescription
+}
+
+func formatMultiline(
+	input string,
+	prefix string,
+	maxLineLen int,
+) string {
+	textLen := maxLineLen - len(prefix)
+
+	// split the input into lines of length textLen
+	lines := strings.Split(input, "\n")
+	var formattedLines []string
+	for _, line := range lines {
+		if len(line) <= textLen {
+			formattedLines = append(formattedLines, line)
+			continue
+		}
+
+		// split the line into multiple lines, don't break words
+		words := strings.Fields(line)
+		var formattedLine string
+		for _, word := range words {
+			if len(formattedLine)+len(word) > textLen {
+				formattedLines = append(formattedLines, formattedLine[1:])
+				formattedLine = ""
+			}
+			formattedLine += " " + word
+		}
+		if formattedLine != "" {
+			formattedLines = append(formattedLines, formattedLine[1:])
+		}
+	}
+
+	// combine lines including indent and prefix
+	var formatted string
+	for _, line := range formattedLines {
+		formatted += prefix + line + "\n"
+	}
+
+	return formatted
+}
+
+func formatParameterValueYAML(value string) string {
+	switch {
+	case value == "":
+		return `""`
+	case strings.Contains(value, "\n"):
+		// specifically used in the javascript processor
+		formattedValue := formatMultiline(value, "            ", 10000)
+		return fmt.Sprintf("|\n%s", formattedValue)
+	default:
+		return fmt.Sprintf(`'%s'`, value)
+	}
+}
```

Comment:
The code for filling out the template has been taken from `processorgen`, except for one small change, which is that values are printed with single quotes (so that they can contain double quotes).